account_analytic_required/models/account.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+# © 2011 Akretion
+# © 2016 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from openerp import _, api, exceptions, fields, models
+from openerp.tools import float_is_zero
+
+
+class AccountAccountType(models.Model):
+    _inherit = "account.account.type"
+
+    @api.model
+    def _get_policies(self):
+        """This is the method to be inherited for adding policies"""
+        return [('optional', 'Optional'),
+                ('always', 'Always'),
+                ('never', 'Never')]
+
+    analytic_policy = fields.Selection(
+        lambda self, *args, **kwargs: self._get_policies(*args, **kwargs),
+        'Policy for analytic account',
+        required=True,
+        default='optional',
+        help="Set the policy for analytic accounts : if you select "
+             "'Optional', the accountant is free to put an analytic account "
+             "on an account move line with this type of account ; if you "
+             "select 'Always', the accountant will get an error message if "
+             "there is no analytic account ; if you select 'Never', the "
+             "accountant will get an error message if an analytic account "
+             "is present.")
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    @api.model
+    def _get_analytic_policy(self, account):
```

Comment:
I think this is there for inheritance purpose that's what I understood from "Extension point" in the docstring.